spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/actions/BaseDeleteOrphanFilesSparkAction.java

```diff
@@ -143,6 +148,24 @@ public BaseDeleteOrphanFilesSparkAction executeDeleteWith(ExecutorService execut
     return this;
   }
 
+  @Override
+  public DeleteOrphanFiles prefixMismatchMode(PrefixMisMatchMode mismatchMode) {
+    this.prefixMismatchMode = mismatchMode;
+    return this;
+  }
+
+  @Override
+  public DeleteOrphanFiles equivalentSchemes(List<String> schemes) {
+    this.equivalentSchemes = schemes;
```

Comment:
These methods need to make a defensive copy of the list. First, that's to ensure that any modification to the list will not change the action's behavior. Second, this is passed directly into an object that gets serialized, so implementations of `List` that are not serializable are problems.

Reply:
I have changed the implementation a bit to use Maps.
We might hit the serialization issues in case of Maps.
We are using HashMap directly in our code(which is serializable), is it ok to leave it as is?

Also, I am not making a copy of the the list.